src/config/middleware.js

```diff
@@ -10,14 +10,41 @@ const checkSessionAndRedirect = (req, res, next) => {
 
   // If the user is on the Home Page and does not have a session, set it
   if (isOnHomepage && !hasSession) {
+    console.log(
+      `user has no session but is on the HP , so lets set the session id`
+    );
     req.session.userSession = {
       startedJourney: true,
     };
   }
 
-  // If the user doesn't have a session and is not on the homepage, redirect to Journey Guard Page
+  // If the user doesn't have a session and is not on the homepage, redirect to Journey Guard Page but if entering on / take them to welcome page
   if (!hasSession && !isOnHomepage) {
```

Comment:
I think you can simplify these conditions. Why? because whatever if the user has a session or not, in you code if the req.path === "/" you redirect him to welcome (conditions l.23 and 36 have the same result)
In reality, you have only 2 scenarios: 
- he has no session and is not on the homepage --> go journeygard
- he is on the homepage. --> go welcome
Double check my comment in case I forget/don't know another scenario.

Reply:
if the user enters the app first time via the root - go to `/welcome`
So If I want this to happen then I need to check the if the user has` !hasSession `and is r`eq.path===" / "` if so redirect to welcome page.

But when I add this , when the user is entering /choose-location to the browser tab for example. The behaviour of the node app is to go to / then go to /choose-location. Which means that it will redirect the path  `/ to  /welcome `, set the session id and then go to /choose-location page and so the session id is set prematurely and the user does not see the journey guard page

Reply:
So I added the code in line 23 to try and catch this scenario but it does not work because it does not only run when the user goes to the root but also whenever the enter a path .

Reply:
I cant just have a condition for this:
he has no session and is not on the homepage --> go journeygard

because if they have no session and go to / or localhost:3000 it will take them to journey guard , which is weird no ?